Sources/_SwiftSyntaxMacros/ExpressionMacro.swift

```diff
@@ -20,3 +24,56 @@ public protocol ExpressionMacro: Macro {
     _ macro: MacroExpansionExprSyntax, in context: MacroEvaluationContext
   ) -> MacroResult<ExprSyntax>
 }
+
+#if canImport(_CompilerPluginSupport)
+extension ExpressionMacro {
+  public static func _kind() -> _CompilerPluginKind {
+    .expressionMacro
+  }
+
+  public static func _rewrite(
+    targetModuleName: UnsafePointer<UInt8>,
+    targetModuleNameCount: Int,
+    filePath: UnsafePointer<UInt8>,
+    filePathCount: Int,
+    sourceFileText: UnsafePointer<UInt8>,
+    sourceFileTextCount: Int,
+    localSourceText: UnsafePointer<UInt8>,
+    localSourceTextCount: Int
+  ) -> (UnsafePointer<UInt8>?, count: Int) {
+    let targetModuleNameBuffer = UnsafeBufferPointer(
+      start: filePath, count: targetModuleNameCount)
+    let targetModuleName = String(
+      decoding: targetModuleNameBuffer, as: UTF8.self)
+    let filePathBuffer = UnsafeBufferPointer(
+      start: filePath, count: filePathCount)
+    let filePath = String(decoding: filePathBuffer, as: UTF8.self)
+    let sourceFileTextBuffer = UnsafeBufferPointer(
+      start: sourceFileText, count: sourceFileTextCount)
+    let sourceFileString = String(decoding: sourceFileTextBuffer, as: UTF8.self)
+    let sourceFileSyntax = Parser.parse(source: sourceFileString)
+    let converter = SourceLocationConverter(
+      file: filePath, tree: sourceFileSyntax)
+    let context = MacroEvaluationContext(
+      moduleName: targetModuleName, sourceLocationConverter: converter)
+    let meePosition = AbsolutePosition(
+      utf8Offset: localSourceText.distance(to: localSourceText))
+    guard let meeStartToken = sourceFileSyntax.token(at: meePosition),
+          let mee = meeStartToken.parent?.as(MacroExpansionExprSyntax.self)
+    else {
+      fatalError("Unable to locate 'MacroExpansionExprSyntax'")
```

Comment:
It makes me pretty nervous to have a `fatalError` in something that will be in the compiler process. In some follow-up PR, can you find a way to report the failure back out of this API?

Reply:
Yes I'll find a way to allow diagnostics to be reported back